### ISSUE TITLE
[FMI deployment] Does PROJECT_DIR need to be visible to the server?

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-server#555]
Does the PROJECT_DIR need to be visible to the fractal user? The Fractal user doesn't put anything there, it just uses sudo cat to copy from there, right? 

I can work around this for the moment, wondering if it's a requirement

If I put the PROJECT_DIR in a place where the server can't see it, project creation fails with:
```
fractal-client-1.1.0a2) [luetjoel@faimsrv01 01_cardio_tiny_dataset]$ . test_create.sh 
removed '/BASE/PATH/fractal-demos/examples/01_cardio_tiny_dataset/.cache/session'
removed directory: '/BASE/PATH/fractal-demos/examples/01_cardio_tiny_dataset/.cache'
INFO:root:name='proj-cardiac-test-create' project_dir='/BASE/PATH/fractal-demos/examples/01_cardio_tiny_dataset/tmp_cardiac-test-create' read_only=False default_dataset_name='default'
ERROR:root:Server returned 422
ERROR:root:Original request: POST http://localhost:8010/api/v1/project/
ERROR:root:Original payload: {"name": "proj-cardiac-test-create", "project_dir": "/BASE/PATH/fractal-demos/examples/01_cardio_tiny_dataset/tmp_cardiac-test-create"}
ERROR:root:Server error message: {'detail': 'Project dir /BASE/PATH/fractal-demos/examples/01_cardio_tiny_dataset/tmp_cardiac-test-create does not exist'}

ERROR:root:Terminating.

```

Even though the actual directory exists, the fractal user just doesn't have permission to see that folder or its base folder.

**Comment by tcompa** [comment]
Agreed, this is wrong.

Out of the current three checks
```python
    # Check that project_dir is an absolute path
    if not os.path.isabs(project.project_dir):
        raise HTTPException(
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
            detail=(
                f"Project dir {project.project_dir} is not an absolute path"
            ),
        )
    # Check that project_dir is an existing path
    if not os.path.exists(project.project_dir):
        raise HTTPException(
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
            detail=(f"Project dir {project.project_dir} does not exist"),
        )
    # Check that project_dir is directory path
    if not os.path.isdir(project.project_dir):
        raise HTTPException(
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
            detail=(f"Project dir {project.project_dir} is not a directory."),
        )
```
only the first one should remain in the current form.
As for the other two checks:
1. In general, they cannot happen without sudo.
2. Broader issue: https://github.com/fractal-analytics-platform/fractal-server/issues/553. This means that possibly this checks will be transformed or moved somewhere else.

**Comment by tcompa** [comment]
With https://github.com/fractal-analytics-platform/fractal-server/commit/c9f7994b767b58f81a9696c9a83b9908d5aa30f6 we are removing the second and third check, which will then be addressed as part of #553.
